idioms/ffi-passing-strings.md

```diff
@@ -0,0 +1,76 @@
+# Passing Strings
+
+## Description
+
+When passing strings to FFI functions, there are four principles that should be followed:
+
+1. Make the lifetime of owned strings as long as possible.
+1. Minimize `unsafe` code during the conversion.
+1. If the C code can modify the string data, use `Vec` instead of `CString`.
+1. Unless the Foreign Function API requires it, the ownership of the string should not transfer to the callee.
```

Comment:
```suggestion
1. Make the lifetime of owned strings as long as possible.
2. Minimize `unsafe` code during the conversion.
3. If the C code can modify the string data, use `Vec` instead of `CString`.
4. Unless the Foreign Function API requires it, the ownership of the string should not transfer to the callee.
```